v8/config/krb5conf.go

```diff
@@ -490,8 +490,9 @@ func (d *DomainRealm) deleteMapping(domain, realm string) {
 }
 
 // ResolveRealm resolves the kerberos realm for the specified domain name from the domain to realm mapping.
-// The most specific mapping is returned.
-func (c *Config) ResolveRealm(domainName string) string {
+// The most specific mapping is returned.  If none is found, the default realm is returned, unless
+// useDefault is false
+func (c *Config) ResolveRealm(domainName string, useDefault bool) string {
```

Comment:
Just noting that this changes a public interface so would require a major version iteration.